src/sql/workbench/contrib/resourceViewer/browser/resourceViewerEditor.ts

```diff
@@ -0,0 +1,191 @@
+import { ResourceViewerTable } from 'sql/workbench/contrib/resourceViewer/browser/resourceViewerTable';
+
+export const CONTEXT_RESOURCE_VIEWER_EDITOR = new RawContextKey<boolean>('inResourceViewerEditor', false);
+
+export interface ResourceViewerTableViewState {
+	scrollTop: number;
+	scrollLeft: number;
+}
+
+export class ResourceViewerEditor extends BaseEditor {
+	public static readonly ID: string = 'workbench.editor.resource-viewer';
+
+	private _container!: HTMLElement;
+	private _actionBar!: Taskbar;
+	private _resourceViewerTable!: ResourceViewerTable;
+	private _stateListener: IDisposable | undefined;
+	private _columnChangeListener: IDisposable | undefined;
+	private _rowCountChangeListener: IDisposable | undefined;
+	private _filterStateChangeListener: IDisposable | undefined;
+
+	private _resourceViewerEditorContextKey: IContextKey<boolean>;
+
+	private _savedTableViewStates = new Map<ResourceViewerInput, ResourceViewerTableViewState>();
+
+	constructor(
+		@ITelemetryService telemetryService: ITelemetryService,
+		@IWorkbenchThemeService themeService: IWorkbenchThemeService,
+		@IInstantiationService private _instantiationService: IInstantiationService,
+		@IContextKeyService private _contextKeyService: IContextKeyService,
+		@IEditorService editorService: IEditorService,
+		@IStorageService storageService: IStorageService
+	) {
+		super(ResourceViewerEditor.ID, telemetryService, themeService, storageService);
+		this._resourceViewerEditorContextKey = CONTEXT_RESOURCE_VIEWER_EDITOR.bindTo(this._contextKeyService);
+
+		if (editorService) {
+			editorService.overrideOpenEditor({
+				open: (editor, options, group) => {
+					if (this.isVisible() && (editor !== this.input || group !== this.group)) {
+						this.saveEditorViewState();
+					}
+					return {};
+				}
+			});
+		}
+	}
+
+	protected createEditor(parent: HTMLElement): void {
+		this._container = document.createElement('div');
+		this._container.className = 'resource-viewer';
+		parent.appendChild(this._container);
+
+		this._createHeader();
+
+		let tableContainer = this.createResourceViewerTable();
+
+		this._container.appendChild(tableContainer);
+	}
+
+	private _createHeader(): void {
+		const header = document.createElement('div');
+		header.className = 'resource-viewer-header';
+		this._container.appendChild(header);
+		this._actionBar = new Taskbar(header);
+
+		this._actionBar.setContent([
+			// TODO - chgagnon add actions
+		]);
+	}
+
+	private createResourceViewerTable(): HTMLElement {
+		let resourceViewerTableContainer = document.createElement('div');
+		resourceViewerTableContainer.className = 'resource-viewer-table monaco-editor';
+		resourceViewerTableContainer.style.width = '100%';
+		resourceViewerTableContainer.style.height = '100%';
+		resourceViewerTableContainer.style.overflow = 'hidden';
+		resourceViewerTableContainer.style.position = 'relative';
+		let theme = this.themeService.getColorTheme();
+		if (theme.type === DARK) {
```

Comment:
Why do you need to do this? the DOM already has this css class on the body element

Reply:
@anthonydresser Removed